uiowa/uiowa
# Classrooms split, module, initial room content type with building dropdown and title

Relates to https://github.com/uiowa/uiowa/issues/5520
Resolves https://github.com/uiowa/uiowa/issues/5895

This PR establishes a connection to the MAUI API Astra building and room data and provides a configuration entity (building) to populate a dropdown reference list on the "room" content type. Finishing the content type is here - https://github.com/uiowa/uiowa/issues/5769

# To Test
```
ddev blt ds --site=classrooms.uiowa.edu && ddev drush @classrooms.local en classrooms_core -y && ddev drush @classrooms.local config:import --source ../config/sites/classrooms.uiowa.edu --partial -y && ddev drush @classrooms.local cim -y && ddev drush @classrooms.local uli node/add/room
```
- `ddev blt ds --site=classrooms.uiowa.edu`
- `ddev drush @classrooms.local en classrooms_core` (to establish the config entity properly)
- `ddev drush @classrooms.local config:import --source ../config/sites/classrooms.uiowa.edu --partial`
- `ddev drush @classrooms.local cim`
- `ddev drush @classrooms.local uli node/add/room`
- See a building id dropdown populated with building names.
- Grab a building id/room id combo from https://api.maui.uiowa.edu/maui/api/pub/registrar/courses/AstraBldgRmCompleteList/list
- The room node title should combine the building name and room.
- General content type creation defaults https://wiki.uiowa.edu/display/drupaldev/Content+type+creation+notes

# Additional testing

- `ddev drush @classrooms.local classrooms_core:buildings` will attempt to import any new buildings into the website. This data is stored in config. You can temporarily delete a building entity and run the drush command to re-import it.
